File: bud-api/src/main/kotlin/com/budapi/model/PlantReference.kt
package com.budapi.model

data class PlantReference(
    var id: String?,
    val name: String,
    val temperature_min: Int,
    val temperature_max: Int,
    val ph_min: Int,
    val ph_max: Int,
    val luminosity_min: Int,
    val luminosity_max: Int,
    val humidity_min: Int,
    val humidity_max: Int,
) {
    constructor() : this("", "", 0, 0, 0,
        0, 0, 0, 0, 0)
}

File: bud_app/lib/model/plant_reference/plant_reference.dart
class PlantReference {
  final String id, name;
  final double humidityMin,
      humidityMax,
      luminosityMin,
      luminosityMax,
      phMin,
      phMax,
      temperatureMin,
      temperatureMax;

  PlantReference(
      this.id,
      this.name,
      this.humidityMin,
      this.humidityMax,
      this.luminosityMin,
      this.luminosityMax,
      this.phMin,
      this.phMax,
      this.temperatureMin,
      this.temperatureMax);

  factory PlantReference.fromJson(dynamic json) {
    return PlantReference(
      json["id"] as String,
      json["name"] as String,
      json["humidity_min"] + 0.0,
      json["humidity_max"] + 0.0,
      json["luminosity_min"] + 0.0,
      json["luminosity_max"] + 0.0,
      json["ph_min"] + 0.0,
      json["ph_max"] + 0.0,
      json["temperature_min"] + 0.0,
      json["temperature_max"] + 0.0,
    );
  }

  static List<PlantReference> plantRefsFromSnapshot(List snapshot) {
    return snapshot.map((data) {
      return PlantReference.fromJson(data);
    }).toList();
  }
}
